experiments/experiments.go

```diff
@@ -0,0 +1,219 @@
+package experiments
+
+import (
+	"context"
+	"crypto/sha1"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/big"
+	"strings"
+	"time"
+
+	"github.com/reddit/baseplate.go/filewatcher"
+	"github.com/reddit/baseplate.go/log"
+)
+
+type Document map[string]*ExperimentConfig
+
+type Experiments struct {
+	watcher *filewatcher.Result
+}
+
+func NewExperiments(ctx context.Context, path string, logger log.Wrapper) (*Experiments, error) {
+	parser := func(r io.Reader) (interface{}, error) {
+		var doc Document
+		err := json.NewDecoder(r).Decode(&doc)
+		if err != nil {
+			return nil, err
+		}
+		return doc, nil
+	}
+	result, err := filewatcher.New(ctx, path, parser, logger)
+	if err != nil {
+		return nil, err
+	}
+	return &Experiments{
+		watcher: result,
+	}, nil
+}
+
+type Bucket struct {
+}
+
+type Experiment interface {
+	UniqueID(map[string]string) string
+	Variant(args map[string]string) (string, error)
+	LogBucketing() bool
+}
+
+func (e *Experiments) Variant(name string, args map[string]string, bucketingEventOverride bool) (string, error) {
+	experiment, err := e.experiment(name)
+	if err != nil {
+		return "", err
+	}
+	return experiment.Variant(args)
```

Comment:
Shouldn't `Variant` also log an event?

Reply:
Yes, I just saw that this branch is missing the event logging. Maybe it would make sense to add this in another PR?

Reply:
Ya this PR is getting pretty big, I think it's fine to do multiple PRs to get this out in manageable chunks rather than trying to get it all in one big PR.